octoprint_costestimation/static/js/costestimation.js

```diff
@@ -214,8 +225,28 @@ $(function() {
             if (self.spoolManager === null){
                 self.settings.settings.plugins.costestimation.useSpoolManager(false);
             }
+            if (self.spoolman === null){
+                self.settings.settings.plugins.costestimation.useSpoolman(false);
+            }
         }
 
+        self.readSpoolmanData = ko.computed(function() {
+            const result = [];
+            for (const spool of self.spoolman.templateData.selectedSpoolsByToolIdx()) {
+                if (spool.spoolId == null) continue;
+                const pulledData = {
+                    cost: spool.spoolData.price || spool.spoolData.filament.price || 0,
+                    weight: spool.spoolData.remaining_weight || 0,
```

Comment:
Using "remaining_weight"  will increase the cost of a print as a spool gets used up.